src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/hooks/useCreateDrawerForm.tsx

```diff
@@ -235,12 +234,19 @@ const useCreateDrawerForm = (
   };
 
   const onChangeStartVM = (checked: boolean) => {
-    setStartVM(checked);
-    updateTabsData((currentTabsData) => {
-      return { ...currentTabsData, startVM: checked };
-    });
+    setVM(
+      produce(vm, (draftVM) => {
+        delete draftVM.spec.runStrategy;
+        draftVM.spec.running = checked;
+      }),
+    );
   };
 
+  useEffect(() => {
```

Comment:
u removed the startVM state , but added this 1 time run hook with disabled liners, I don't like this practice.
I don't think this is needed at all, u trying to achieve that if no runStrategy mark  the start VM checkbox in the end?

what will happen if you put in line 269 startVM: vm?.spec?.running && !VM?.spec?.runStrayrgy or some combination of this? can this hook be removed?

Reply:
okay. I was trying to set running true at the beginning just like before with the useState

Reply:
So i removed the useEffect. This means that the checkbox will reflect the template configuration. If the template has a VM with running: false, the checkbox at the beginning will be false. Is it okay?